### PR TITLE
Doc updates

### PR DESCRIPTION
**Pull request by mhaskel**

### CODE REVIEW

**Comment by mhaskel** [comment]
@daenney I _think_ this gets the README up to being accurate with classes/defines/parameters and examples. It could certainly use more polish, but I'd probably defer to our docs team for that.

I think our big missing pieces at this point is an upgrade guide, and updating things in `examples`. We're getting close!

so close

**Comment by daenney** [comment]
Such cuteness!

File: README.md [hunk 1]
```diff
@@ -54,38 +41,11 @@ class { 'apt':
...
-    blacklist   => [],
-    update      => '1',
-    download    => '1',
-    upgrade     => '1',
-    autoclean   => '7',
-  }
-  ```
-  
-* `apt::update`: Runs `apt-get update`, updating the list of available packages and their versions without installing or upgrading any packages. The update runs on the first Puppet run after you include the class, then whenever `notify  => Exec['apt_update']` occurs; i.e., whenever config files get updated or other relevant changes occur. If you set the `always_apt_update` parameter to 'true', the update runs on every Puppet run.
+* `apt::update`: Runs `apt-get update`, updating the list of available packages and their versions without installing or upgrading any packages. The update runs on the first Puppet run after you include the class, then whenever `notify  => Exec['apt_update']` occurs; i.e., whenever config files get updated or other relevant changes occur. If you set `update['frequency']` to 'true', the update runs on every Puppet run.
```
Comment (daenney) [inline README.md]: Careful with the quotes here `'true'` is the string, we need it set to `true` the Boolean.

File: README.md [hunk 2]
```diff
@@ -242,18 +163,20 @@ apt::sources:
...
-    key: '9AA38DCD55BE302B'
-    key_server: 'subkeys.pgp.net'
+    key:
+      id: '9AA38DCD55BE302B'
+      server: 'subkeys.pgp.net'
     pin: '-10'
-    include_src: 'true'
-    include_deb: 'true'
+    include:
+      src: 'true'
```
Comment (daenney) [inline README.md]: Careful with the quotes here `'true'` is the string, we need it set to `true` the Boolean.

File: README.md [hunk 3]
```diff
@@ -242,18 +163,20 @@ apt::sources:
...
+      deb: 'true'
```
Comment (daenney) [inline README.md]: Careful with the quotes here `'true'` is the string, we need it set to `true` the Boolean.

File: README.md [hunk 4]
```diff
@@ -261,64 +184,48 @@ apt::sources:
...
-* `key_server`: The keyserver from where to fetch our GPG key. It can either be a domain name or URL. It defaults to undef which results in apt_key's default keyserver being used, currently `keyserver.ubuntu.com`.
-* `key_options`: Additional options to pass on to `apt-key adv --keyserver-options`.
+* `id`: Is a GPG key ID or full key fingerprint. This value is validated with a regex enforcing it to only contain valid hexadecimal characters, be precisely 8 or 16 hexadecimal characters long and optionally prefixed with 0x for key IDs, or 40 hexadecimal characters long for key fingerprints.
+* `content`: This parameter can be used to pass in a GPG key as a string in case it cannot be fetched from a remote location and using a file resource is for other reasons inconvenient.
+* `source`: This parameter can be used to pass in the location of a GPG key. This URI can take the form of a `URL` (ftp, http or https) and a `path` (absolute path to a file on the target system).
+* `server`: The keyserver from where to fetch our GPG key. It can either be a domain name or URL. It defaults to undef which results in apt_key's default keyserver being used, currently `keyserver.ubuntu.com`.
```
Comment (daenney) [inline README.md]: It doesn't default to `undef` anymore, we set it explicitly to `keyserver.ubuntu.com`.

File: README.md [hunk 5]
```diff
@@ -334,39 +241,41 @@ apt::sources:
...
 
-`$ puppet parser validate tests/puppetlabs-apt.pp`
+####apt::setting
 
-If you receive no output from that command, it means nothing is wrong. Then, apply the code:
+* `priority`: Integer or zero-padded integer setting the file priority. Defaults to 50.
+* `ensure`: Whether to add or remove the file. Valid values are 'present', 'absent', and 'file'. Defaults to file.
+* `source`: The source for the file. Exactly one of `content` and `source` must be specified.
+* `content`: The content for the file. Exactly one of `content` and `source` must be specified.
+* `notify_update`: Boolean for whether or not this `apt::setting` should notify `Class['apt::update']`. Defaults to true.
```
Comment (daenney) [inline README.md]: Perhaps lets say here "Boolean for whether or not this `apt::setting` should trigger an `apt-get update`. Defaults to true"

File: README.md [hunk 6]
```diff
@@ -334,39 +241,41 @@ apt::sources:
...
+####apt::source
 
-The above example uses a smoke test to lay out a resource declaration and apply it on your system. In production, you might want to declare your Apt sources inside the classes where they’re needed.
+* `comment`: Add a comment to the apt source file.
+* `ensure`: Allows you to remove the apt source file. Can be 'present' or 'absent'.
+* `location`: The URL of the apt repository. Defaults to undef. Required unless `ensure => 'absent'`.
+* `release`: The distribution of the apt repository. Defaults to fact 'lsbdistcodename'.
+* `repos`: The component of the apt repository. This defaults to 'main'.
+* `include`: Hash to configure include options. Valid keys are:
+  * 'deb': References a Debian distribution's binary package.
```
Comment (daenney) [inline README.md]: Mention this defaults to true.

File: README.md [hunk 7]
```diff
@@ -334,39 +241,41 @@ apt::sources:
...
+  * 'src': Enable the deb-src type, references a Debian distribution's source code in the same form as the include_deb type. A deb-src line is required to fetch source indexes.
```
Comment (daenney) [inline README.md]: Mention this defaults to false.

File: README.md [hunk 8]
```diff
@@ -334,39 +241,41 @@ apt::sources:
...
+* `key`: Add key from source. Takes either a string or a hash. If a string, the value will be passed to `id` in the `apt::key`. If a hash, valid keys are:
+  * 'id': See `id` in `apt::key`. Required if a hash is specified.
+  * 'server': See `server` in `apt::key`
+  * 'content': See `content` in `apt::key`
+  * 'source': See `source` in `apt::key`
+  * 'options': See `options` in `apt::key`
+* `pin`: See apt::pin. Defaults to false.
+* `architecture`: can be used to specify for which architectures information should be downloaded. If this option is not set all architectures defined by the APT::Architectures option will be downloaded. Defaults to 'undef' which means all. Example values can be 'i386' or 'i386,alpha,powerpc'.
```
Comment (daenney) [inline README.md]: Careful with the quotes `'undef'` is not `undef`.

File: README.md [hunk 9]
```diff
@@ -334,39 +241,41 @@ apt::sources:
...
+* `allow_unsigned`: can be set to indicate that packages from this source are always authenticated even if the Release file is not signed or the signature can't be checked. Defaults to false. Can be 'true' or 'false'.
```
Comment (daenney) [inline README.md]: Careful with the quotes here `'true'` and `'false'` are strings, we need it set to `true` or `false` the Booleans.

**Comment by daenney** [comment]
L22 in the README still mentions the old `purge_*` parameters, we should remove that.

File: README.md [hunk 10]
```diff
@@ -21,7 +19,7 @@ The apt module automates obtaining and installing software packages on \*nix sys
 * System repositories
 * Authentication keys
 
-**Note**: Setting the apt module's `purge_sources_list` and `purge_sources_list_d` parameters to 'true' will **destroy** any existing content that was not declared with Puppet. The default for these parameters is 'false'.
+**Note**: Setting the apt module's `purge['sources.list']` and `purge['sources.list.d']` parameters to `true` will **destroy** any existing content that was not declared with Puppet. The default for these parameters is `false`.
```
Comment (daenney) [inline README.md]: Actually, the default is `true`: https://github.com/puppetlabs/puppetlabs-apt/blob/next/manifests/params.pp#L71-L76 :smile:.

File: README.md [hunk 11]
```diff
@@ -54,38 +41,11 @@ class { 'apt':
 
 * `apt`: Main class, provides common resources and options. Allows Puppet to manage your system's sources.list file and sources.list.d directory, but it does its best to respect existing content.
 
-  If you declare your apt class with `purge_sources_list`, `purge_sources_list_d`, `purge_preferences` and `purge_preferences_d` set to 'true', Puppet will unapologetically purge any existing content it finds that wasn't declared with Puppet.
+  If you declare your apt class with any of the options in the `purge` hash set to 'true', Puppet will unapologetically purge any existing content it finds that wasn't declared with Puppet.
```
Comment (daenney) [inline README.md]: So they're already set to `true`.

File: README.md [hunk 12]
```diff
@@ -261,64 +184,48 @@ apt::sources:
...
+    * 'always': Runs update at every Puppet run.
+    * 'daily': Runs update daily; that is, `apt-get update` runs if the value of `apt_update_last_success` is less than current epoch time - 86400. If the exec resource `apt_update` is notified, `apt-get update` runs regardless of this value. 
+    * 'weekly': Runs update weekly; that is, `apt-get update` runs if the value of `apt_update_last_success` is less than current epoch time - 604800. If the exec resource `apt_update` is notified, `apt-get update` runs regardless of this value. 
+    * 'reluctantly': Only runs `apt-get update` if the exec resource `apt_update` is notified. This is the default setting.  
+  * 'timeout': Overrides the exec timeout in seconds for `apt-get update`. Defaults to exec default (300).
+  * 'tries': Sets how many times to attempt running `apt-get update`. Use this to work around transient DNS and HTTP errors. By default, the command runs only once.
+* `purge`: Hash to configure various purge settings. Valid keys are:
+  * 'sources.list': If set to 'true', Puppet purges all unmanaged entries from sources.list. Accepts 'true' or 'false'. Defaults to 'false'.
```
Comment (daenney) [inline README.md]: Is set to true.

File: README.md [hunk 13]
```diff
@@ -261,64 +184,48 @@ apt::sources:
...
+  * 'sources.list.d': If set to 'true', Puppet purges all unmanaged entries from sources.list.d. Accepts 'true' or 'false'. Defaults to 'false'.
```
Comment (daenney) [inline README.md]: Defaults to true.

File: README.md [hunk 14]
```diff
@@ -261,64 +184,48 @@ apt::sources:
...
+  * 'preferences.list': If set to 'true', Puppet purges all unmanaged entries from preferences.list. Accepts 'true' or 'false'. Defaults to 'false'.
```
Comment (daenney) [inline README.md]: Defaults to true.

File: README.md [hunk 15]
```diff
@@ -261,64 +184,48 @@ apt::sources:
...
+  * 'preferences.list.d': If set to 'true', Puppet purges all unmanaged entries from preferences.list.d. Accepts 'true' or 'false'. Defaults to 'false'.
```
Comment (daenney) [inline README.md]: Defaults to true.

**Pull request merged**